kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -268,6 +268,13 @@ impl<'tcx> GotocCtx<'tcx> {
             }};
         }
 
+        // If the intrinsic is defined on generic types, its trimmed name will
+        // include type arguments (e.g., `arith_offset::<u8>`). In that case, we
+        // remove them to simplify the match below.
+        let intrinsic_split = intrinsic.split_once("::");
```

Comment:
This doesn't look robust to me. Can we perhaps monomorphize the instance to guarantee the removal of the generic?

Reply:
I see. I was concerned about the possibility of having multiple `::` separators as is the case with other functions (e.g. `core::mem::...`, but I'm not sure whether this can occur. Perhaps using `rsplit_once` would be better to use to make sure we're removing the last occurrence? But again, if multiple separators are possible, then using `rsplit_once` may incorrectly split a non-generic portion.

Reply:
Multiple `::` separators shouldn't be possible because we're getting a trimmed name. We can't really do much about that with `split_once` or `rsplit_once`, but we can count the number of occurrences (with `matches`) and assert. I've added this check now.

Reply:
Cool, thanks.

Reply:
Maybe we should add a better mechanism to do this check in StableMIR. It does look very fragile. @adpaco-aws, can you please create an issue here: https://github.com/rust-lang/project-stable-mir/issues

Reply:
Yes, opened https://github.com/rust-lang/project-stable-mir/issues/65